src/java/org/apache/cassandra/cql3/restrictions/ClusteringColumnRestrictions.java

```diff
@@ -197,9 +170,9 @@ public boolean needFiltering()
                 return true;
 
             if (!restriction.isSlice())
-                position = restriction.getLastColumn().position() + 1;
+                position = restriction.lastColumn().position() + 1;
         }
-        return hasContains();
+        return false;
```

Comment:
So this is safe because `handleInFilter()` handles the "contains" case anyway? (Was the `hasContains()` check actually redundant before, since `handleInFilter()` returned `true` on the "contains" case?)